spec.bs

```diff
@@ -2664,6 +2695,82 @@ To <dfn>fetch trusted signals</dfn> given a [=URL=] |url|, an [=origin=] |script
+  ::  «`Accept`: `message/ad-auction-trusted-signals-response`»
+  :   [=request/client=]
+  ::  `null`
+  :   [=request/mode=]
+  ::  "`cors`"
+  :   [=request/referrer=]
+  ::  "`no-referrer`"
+  :   [=request/credentials mode=]
+  ::  "`omit`"
+  :   [=request/redirect mode=]
+  ::  "`error`"
+  :   [=request/service-workers mode=]
+  ::  "`none`"
+  :   [=request/policy container=]
+  ::  A new [=policy container=] whose [=policy container/IP address space=] is |policyContainer|'s
+    [=policy container/IP address space=]
+
+1. Let |signals| be null.
+1. Let |dataVersion| be an empty [=ordered map=], whose [=map/keys=] are [=strings=] and [=map/values=] are integers.
+1. Let |perInterestGroupData| be an [=ordered map=] whose [=map/keys=] are [=interest group/name=] [=strings=]
+  and whose [=map/values=] are [=bidding signals per interest group data=].
+1. [=Fetch=] |request| with [=fetch/useParallelQueue=] set to true, and
+  [=fetch/processResponseConsumeBody=] set to the following steps given a [=response=] |response|
+  and null, failure, or a [=byte sequence=] |responseBody|:
+  1. If [=validate fetching response=] with |response|, |responseBody| and
+    "`message/ad-auction-trusted-signals-response`" returns false, set |signals| to failure and return.
+  1. Let |resultList| be the result of deserializing |responseBody| using |context|. The
+    deserialization method may follow that described in
+    [Section 2.3.6 of the Protected Audience Key Value Services](https://privacysandbox.github.io/draft-ietf-protected-audience-key-value-service/draft-ietf-protected-audience-key-value-services.html#name-parsing-a-response).
+  1. [=list/For each=] |result| in |resultList|:
+    1. If |isBiddingSignal| is true:
+      1. [=map/For each=] |name| → |value| in |result|["interestGroupNames"]:
+        1. If |indexMap|[|name|] does not equal to |result|["index"], set |signals| to failure and return.
+        1. [=map/Set=] |perInterestGroupData|[|name|] to |value|.
+        1. If |result| [=map/contain=] "dataVersion":
+          1. If |result|["dataVersion"] is not an integer, or is less than 0 or more than
+            2<sup>32</sup>&minus;1, set |signals| to failure and return.
+          1. Otherwise, set |dataVersion|[|name|] to |result|["dataVersion"].
+      1. [=map/For each=] |key| → |value| in |result|["keys"]:
+        1. Set |signals|[|key|] to |value|.
+    1. Otherwise:
+      1. [=map/For each=] |url| → |value| in |result|["renderUrls"]:
+        1. Set |signals|[|url|] to |value|.
+        1. If |result| [=map/contains=] `"dataVersion"`:
+          1. If |result|["dataVersion"] is not an integer, or is less than 0 or more than
+            2<sup>32</sup>&minus;1, set |signals| to failure and return.
+          1. Otherwise, set |dataVersion|[|url|] to |result|["dataVersion"].
+      1. [=map/For each=] |url| → |value| in |result|["adComponentRenderUrls"]:
+        1. Set |signals|[|url|] to |value|.

```

Comment:
is it possible that an adComponentRenderUrl is same as an renderUrl? If so, this would overwrite the old |signals|[url| value you set above, is that expected?

Reply:
This is followed by the same call flow of current trusted signals logic, which should not have the same url between render and ad component render ones.